Tools/AnnotationTool/UserInterface/ColorPickerActivator.cs

```diff
@@ -0,0 +1,102 @@
+#if UNITY_EDITOR
+using System;
+using System.Collections;
+using UnityEditor.Experimental.EditorVR.Extensions;
+using UnityEditor.Experimental.EditorVR.UI;
+using UnityEditor.Experimental.EditorVR.Utilities;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public class ColorPickerActivator : MonoBehaviour, IPointerClickHandler, IPointerEnterHandler, IPointerExitHandler
+{
+	[SerializeField]
+	Transform m_TargetScale;
+
+	[SerializeField]
+	Transform m_Icon;
+
+	[SerializeField]
+	GameObject m_Undo;
+
+	[SerializeField]
+	GameObject m_Redo;
+
+	Coroutine m_HighlightCoroutine;
+	GradientButton m_UndoButton;
+	GradientButton m_RedoButton;
+
+	public Transform rayOrigin { private get; set; }
+	public Action<Transform> showColorPicker { private get; set; }
+	public Action hideColorPicker { private get; set; }
+
+	public event Action undoButtonClick
+	{
+		add { m_UndoButton.click += value; }
```

Comment:
are get/set not working here or are we just being explicit about what the user can do with these properties?  Compiler is usually smart enough to handle the add/remove logic automatically

Reply:
Events don't have get/set.  The idea here is to avoid exposing the UndoButton itself publicly.

Reply:
Yep, I misremembered here